### PR TITLE
TST/CLN: Remove unnecessary pyarrow version checking

### PR DESCRIPTION
**Pull request by mroeschke**
I think this shouldn't be necessary since 7.0 is the minimum version supported now.

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -569,11 +569,8 @@ def argsort(
     ) -> np.ndarray:
         order = "ascending" if ascending else "descending"
         null_placement = {"last": "at_end", "first": "at_start"}.get(na_position, None)
-        if null_placement is None or pa_version_under7p0:
-            # Although pc.array_sort_indices exists in version 6
-            # there's a bug that affects the pa.ChunkedArray backing
-            # https://issues.apache.org/jira/browse/ARROW-12042
-            fallback_performancewarning("7")
+        if null_placement is None:
+            fallback_performancewarning()
```
Comment (jbrockmendel) [inline pandas/core/arrays/arrow/array.py:573]: is the warning still necessary?

**Review by jbrockmendel — APPROVED** [review]
LGTM pending green

File: pandas/core/arrays/arrow/array.py (hunk 1 above)
Comment (mroeschke) [reply pandas/core/arrays/arrow/array.py:573]: Good catch. I think now this can raise an error directly

**Comment by jbrockmendel** [comment]
FAILED pandas/tests/indexes/test_common.py::test_sort_values_invalid_na_position[string-pyarrow-None] - AssertionError: Did not see expected warning of class 'PerformanceWarning'
FAILED pandas/tests/indexes/test_common.py::test_sort_values_invalid_na_position[string-pyarrow-middle] - AssertionError: Did not see expected warning of class 'PerformanceWarning'

**Comment by mroeschke** [comment]
Looks like the (some?) failures are due to https://github.com/pandas-dev/pandas/pull/51483 (should have been more careful with merging after being stricter with the warnings)

**Comment by phofl** [comment]
Can you remove my tm.assert_produces_warning when you are ready here?

**Comment by mroeschke** [comment]
> Can you remove my tm.assert_produces_warning when you are ready here?

Sure, done in the last commit.

Going to merge and backport just in case 2.0 has any PRs that impact `test_arrow.py`

**Pull request merged**